### DISCUSSION TITLE
Set the area automatically for automations/scenes/(scripts) by the devices used in automation

### DISCUSSION THREAD

**Discussion by ahti123** [home-assistant/frontend#16106] [Automations / scripts / scenes]
It is an annoying extra step(s) to manually set the area for the automation whereas you most likely create the automation for specific single area.
And there are too many steps to set the area:
0. create automation (you can't set the area in the editor itself?)
1. save the automation, back in the list
2. look it up from the (long) list of automations
3. three dots -> information -> settings - only there you can change the area

With any basic number of automations you end up with many and it would be really useful to set the area more easily.

**Comment by frenck** [comment]
We can’t set the area automatically. Any item can be physically only in a single area. All the types you mentioned can and will effect multiple areas. Picking one at random is not a good idea either imho.

**Reply by ahti123** [reply]
Yes, I can see that but if your automation uses devices from a single area only?
Аnother solution would be to bring area selection to the new automation edit screen. Finding it from the list afterwards and information and settings is not very usable.
